Update SCE file contents to include processed SCE objects (#100)
AlexsLemonade/scpca-docs: docs/sce_file_contents.md

| `subsets_mito_percent`  | Percent of all UMI counts assigned to mitochondrial genes                                                                                                                                     |
| `total`                 | Total UMI count for RNA-seq data and any alternative experiments (i.e., CITE-seq)                                                                                                             |

> The following are additional per-cell data columns included in both the `filtered` and `processed` objects.
> The included metrics were calculated by [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.

Reviewer: My initial edit here was intended to make the statement about metrics less definitive about the source of the data, but I left in the "were"... But then I realize that even with my intended change, that left an unqualified "these", so here I am attempting once again to say what I mean.

```suggestion
The following additional per-cell data columns are included in both the `filtered` and `processed` objects.
These columns include metrics calculated by [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
```

| `subsets_mito_percent`  | Percent of all UMI counts assigned to mitochondrial genes                                                                                                                                     |
| `total`                 | Total UMI count for RNA-seq data and any alternative experiments (i.e., CITE-seq)                                                                                                             |

The following are additional per-cell data columns included in both the `filtered` and `processed` objects.

> These metrics were calculated by using [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
> Cells with a high likelihood of being compromised as calculated by `miQC` (greater than 0.75) and cells that do not pass a minimum number of unique genes detected threshold of 200 are removed from the counts matrix found in the `processed` objects.

Reviewer: I reworded somewhat. I'm not sure we need the detail about the filtering here: I moved the link to the processed gene expression data section up here for brevity. (So line 59 should be deleted)

Secondary thought: we should rename `ccdl_filter` to `scpca_filter`.

```suggestion
These include metrics were calculated by [`miQC`](https://bioconductor.org/packages/release/bioc/html/miQC.html), a package that jointly models proportion of reads belonging to mitochondrial genes and number of unique genes detected to predict low-quality cells.
We also include the filtering results used for the creation of the `processed` objects. 
See the description of the {ref}`processed gene expression data <processing_information:Processed gene expression data>` for more information on filtering performed to obtain the `_processed.rds` file.
```

Reviewer: Filed https://github.com/AlexsLemonade/scpca-nf/pull/239 for the secondary thought... We should update all docs here if that is approved.

| `prob_compromised`      | Probability that a cell is compromised (i.e., dead or damaged), as calculated by `miQC`                                                                                                       |
| `miQC_pass`             | Indicates whether the cell passed the default miQC filtering. `TRUE` is assigned to cells with a low probability of being compromised (`prob_compromised` < 0.75) or [sufficiently low mitochondrial content](https://bioconductor.org/packages/release/bioc/vignettes/miQC/inst/doc/miQC.html#preventing-exclusion-of-low-mito-cells).  |

> For the `_processed.rds` file only, an additional column named, `ccdl_filter` will be included in the `colData`.

Reviewer: Did we decide not to add this back to the filtered data? I thought that we had, according to: 
https://github.com/AlexsLemonade/scpca-nf/blob/f797157dff44804500774f164cc518017c0871e8/bin/post_process_sce.R#L147-L148
and 
https://github.com/AlexsLemonade/scpca-nf/blob/f797157dff44804500774f164cc518017c0871e8/modules/sce-processing.nf#L126-L134

Author: Yes you are right! I'll update this.

| `transcript_type`   | Transcripts included in gene counts: `spliced` for single-cell samples and `unspliced` for single-nuclei                       |
| `miQC_model`        | The model object that `miQC` fit to the data and was used to calculate `prob_compromised`. Only present for `filtered` objects |
| `filtering_method`  | The method used for cell filtering. One of `emptyDrops`, `emptyDropsCellRanger`, or `UMI cutoff`. Only present for `filtered` objects |
| `umi_cutoff`        | The minimum UMI count per cell used as a threshold for removing empty droplets. Only present for `filtered` objects where the `filtering_method` is `UMI cutoff` |
| `prob_compromised_cutoff`        | The minimum cutoff for the probability of a cell being compromised, as calculated by `miQC`. Only present for `filtered` objects |
| `ccdl_filter_method`        | Method used by the Data Lab to filter low quality cells prior to normalization. Either `miQC` or `Minimum_gene_cutoff`. Only present for `processed` objects |

> | `min_gene_cutoff`        | The minimum cutoff for the number of unique genes detected per cell. Only present for `processed` objects |

Reviewer: I believe these are both in the filtered object as well.
https://github.com/AlexsLemonade/scpca-nf/blob/f797157dff44804500774f164cc518017c0871e8/bin/post_process_sce.R#L68-L85

| `normalization`        | The method used for normalization of raw counts. Either `deconvolution`, described in [Lun, Bach, and Marioni (2016)](https://doi.org/10.1186/s13059-016-0947-7), or  `log-normalization`. Only present for `processed` objects |

> | `highly_variable_genes`        | A list of highly variable genes detected in the library, determined using `scran::modelGeneVar` and `scran::getTopHVGs`. Only present for `processed` objects |

Reviewer: I don't think we want to say that we "determined" the HVGs, more how they were used. They are not a definitive list, to be sure
```suggestion
| `highly_variable_genes`        | A list of highly variable genes used for dimensionality reduction, determined using `scran::modelGeneVar` and `scran::getTopHVGs`. Only present for `processed` objects |
```

### Dimensionality reduction results

> In the RDS file containing the processed `SingleCellExperiment` object only (`_processed.rds`), the `reducedDim` slot of the object will be occupied with both principal component analysis (PCA) and UMAP results.

Reviewer: I added quotes here as these are the actual names in the slots. 
```suggestion
In the RDS file containing the processed `SingleCellExperiment` object only (`_processed.rds`), the `reducedDim` slot of the object will be occupied with both principal component analysis (`PCA`) and `UMAP` results.
```

For all other files, the `reducedDim` slot will be empty as no dimensionality reduction was performed.

> PCA results were calculated using `scater::runPCA()`, specifying to only use the highly variable genes stored in the object in `metadata(sce)$highly_variable_genes`.

Reviewer: Do we want to specify the number of PCs (I don't think we need to, but I was thinking about it)
```suggestion
PCA results were calculated using `scater::runPCA()`, using only highly variable genes.
The list of highly variable genes used was selected using `scran::modelGeneVar` and `scran::getTopHVGs`, and is stored in the SCE object in `metadata(sce)$highly_variable_genes`.
```

> The below command can be used to access the PCA results:

Reviewer: ```suggestion
The following command can be used to access the PCA results:
```

```r
reducedDim(sce, "PCA")
```

UMAP results were calculated using `scater::runUMAP()`, with the PCA results as input rather than the full gene expression matrix.

> The below command can be used to access the UMAP results:
> 
> ```r
> reducedDim(sce,"UMAP")
> ```

Reviewer: ```suggestion
The following command can be used to access the UMAP results:

```r
reducedDim(sce, "UMAP")
```
```
